src/Microsoft.AspNetCore.Rewrite/UrlRewriteBuilder.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+using Microsoft.AspNetCore.Rewrite.RuleAbstraction;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore.Rewrite
+{
+    /// <summary>
+    /// The builder to a list of rules for <see cref="UrlRewriteOptions"/> and <see cref="UrlRewriteMiddleware"/> 
+    /// </summary>
+    public static class UrlRewriteBuilder
+    {
+
+        /// <summary>
+        /// Imports rules from a mod_rewrite file and adds the rules to current rules. 
+        /// </summary>
+        /// <param name="options">The UrlRewrite options.</param>
+        /// <param name="filePath">The path to the file containing mod_rewrite rules.</param>
+        public static UrlRewriteOptions ImportFromModRewrite(this UrlRewriteOptions options, string filePath)
+        {
+            if (string.IsNullOrEmpty(filePath))
+            {
+                throw new ArgumentException(nameof(filePath));
+            }
+            using (var stream = File.OpenRead(filePath))
```

Comment:
This should be relative to the application base. Use IHostingEnvironment.ContentRootPath or ContentRootFileProvider